src/@apollo/client/cache/core/types/ApolloClient__Cache_Core_Types_DataProxy.res

```diff
@@ -26,23 +34,34 @@ module Query = {
     query: t.query,
     variables: t.variables->serializeVariables->mapJsVariables,
     id: t.id,
+    // returnPartialData: t.returnPartialData,
+    optimistic: t.optimistic,
+    canonizeResults: t.canonizeResults,
   }
 }
 
-module Fragment = {
+module ReadFragmentOptions = {
   module Js_ = {
     // interface Fragment<TVariables> {
     //   id: string;
     //   fragment: DocumentNode;
     //   fragmentName?: string;
     //   variables?: TVariables;
     // }
+    // export interface ReadFragmentOptions<TData, TVariables> extends Fragment<TVariables, TData> {
+    //   returnPartialData?: boolean;
+    //   optimistic?: boolean;
+    //   canonizeResults?: boolean;
+    // }
     type t = {
       id: string,
       fragment: Graphql.documentNode,
       // We don't allow optional variables because it's not typesafe
       // variables: 'jsVariables,
       fragmentName: option<string>,
+      // returnPartialData: option<bool>,
```

Comment:
probably want to create an issue to track progress on this from `graphql-ppx`?

Reply:
Yes, that makes sense. The implementation I think will be rather ugly which is why it's been avoided to this point. Jaap pointed out that the partialData story is changing in v4.0+ (#111) so we may have the ability to do handle this sooner than later.